src/grammar.lalrpop

```diff
@@ -579,6 +583,76 @@ Todo: UntypedExpr = {
     }
 }
 
+Bitstring: UntypedExpr = {
+    <s:@L> "<<" <elems:Comma<BinSegment>> ">>" <e:@L> => {
+        UntypedExpr::Bin {
+            location: location(s, e),
+            elems: elems,
+        }
+    }
+}
+
+BinSegment: UntypedExprBinSegment = {
+    <s:@L> <v:SimpleExpr> <options:("::" <HyphensNoTrailing<BinSegmentOption>>)?>  <e:@L> =>
```

Comment:
Could we use `:` rather than `::`? To match type annotations for values.

Reply:
Certainly. I actually went for `::` specifically to not match normal types, haha. I can see the arguments both ways though, and would be happy with either.

Reply:
Perhaps we should ask the wider group?

Reply:
Can do